GLTFSDK.Samples/Serialize/Source/main.cpp

```diff
@@ -0,0 +1,294 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <GLTFSDK/GLTF.h>
+#include <GLTFSDK/BufferBuilder.h>
+#include <GLTFSDK/GLTFResourceWriter.h>
+#include <GLTFSDK/GLBResourceWriter.h>
+#include <GLTFSDK/IStreamWriter.h>
+#include <GLTFSDK/Serialize.h>
+
+// Replace this with <filesystem> (and use std::filesystem rather than
+// std::experimental::filesystem) if your toolchain fully supports C++17
+#include <experimental/filesystem>
+
+#include <fstream>
+#include <sstream>
+#include <iostream>
+
+#include <cassert>
+#include <cstdlib>
+
+using namespace Microsoft::glTF;
+
+namespace
+{
+    // The glTF SDK is decoupled from all file I/O by the IStreamWriter (and IStreamReader)
+    // interface(s) and the C++ stream-based I/O library. This allows the glTF SDK to be used in
+    // sandboxed environments, such as WebAssembly modules and UWP apps, where any file I/O code
+    // must be platform or use-case specific.
+    class StreamWriter : public IStreamWriter
+    {
+    public:
+        StreamWriter(std::experimental::filesystem::path pathBase) : m_pathBase(std::move(pathBase))
+        {
+            assert(m_pathBase.has_root_path());
+        }
+
+        // Resolves the relative URIs of any external resources declared in the glTF manifest
+        std::shared_ptr<std::ostream> GetOutputStream(const std::string& filename) const override
+        {
+            // In order to construct a valid stream:
+            //1. The filename argument will be encoded as UTF-8 so use filesystem::u8path to
```

Comment:
(minor) would be nice to have spaces here

Reply:
I've added a few spaces to improve readability.